src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,122 @@
-import { useState } from 'react';
+import { SyntheticEvent, useState } from 'react';
+import { Movie } from '../../types/Movie';
 import { TextField } from '../TextField';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+type Props = {
+  onAdd: (movie: Movie) => void
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [count, setCount] = useState(0);
+
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [imgUrl, setImgUrl] = useState('');
+  const [imdbUrl, setImdbUrl] = useState('');
+  const [imdbId, setImdbId] = useState('');
+  const [validation, setValidation] = useState(true);
+  const isEmpty = !!title.trim() && !!imgUrl.trim()
+  && !!imdbUrl.trim() && !!imdbId.trim();
+
+  const urlRegex = new RegExp(`${
+    /(?:(?:(https?|ftp):)?\/\/)/.source
+  }${/(?:([^:\n\r]+):([^@\n\r]+)@)?/.source
+  }${/(?:(?:www\.)?([^/\n\r]+))/.source
+  }${/(\/[^?\n\r]+)?/.source
+  }${/(\?[^#\n\r]*)?/.source
+  }${/(#?[^\n\r]*)?/.source}`);
+
+  const urlValidation = (value: string) => {
+    if (!urlRegex.test(value)) {
+      setValidation(false);
+    } else {
+      setValidation(true);
+    }
+  };
+
+  const handleAddMovie = (event: SyntheticEvent) => {
```

Comment:
Try to rewrite this way
```suggestion
  const handleAddMovie = (event: React.FormEvent<HTMLFormElement>) => {
```